application/src/main/kotlin/com/studentcenter/weave/application/user/service/util/impl/strategy/AppleOpenIdTokenResolveStrategy.kt

```diff
@@ -1,13 +1,45 @@
 package com.studentcenter.weave.application.user.service.util.impl.strategy
 
+import com.studentcenter.weave.application.common.properties.OpenIdProperties
 import com.studentcenter.weave.application.user.vo.UserTokenClaims
+import com.studentcenter.weave.domain.user.enums.SocialLoginProvider
+import com.studentcenter.weave.domain.user.vo.Nickname
+import com.studentcenter.weave.support.common.vo.Email
+import com.studentcenter.weave.support.security.jwt.util.JwtTokenProvider
+import com.studentcenter.weave.support.security.jwt.vo.JwtClaims
 import org.springframework.stereotype.Component
+import java.net.URL
 
 @Component
-class AppleOpenIdTokenResolveStrategy : OpenIdTokenResolveStrategy {
+class AppleOpenIdTokenResolveStrategy(
+    private val openIdProperties: OpenIdProperties
+) : OpenIdTokenResolveStrategy {
+
+    val emailDelimiter: String = "@"
 
     override fun resolveIdToken(idToken: String): UserTokenClaims.IdToken {
-        TODO("Not yet implemented")
+        val jwksUri =
+            URL(this.openIdProperties.providers.getValue(SocialLoginProvider.APPLE).jwksUri)
+        val result: JwtClaims = JwtTokenProvider.verifyJwksBasedToken(idToken, jwksUri).getOrThrow()
+
+        val email = result.customClaims["email"] as String
+
+        return UserTokenClaims.IdToken(
+            nickname = Nickname(extractNicknameFromEmail(email)),
+            email = Email(email),
+        )
     }
 
+    private fun extractNicknameFromEmail(email: String): String {
+        val nickname: String = email.substringBefore(emailDelimiter)
+
+        return adjustNickname(nickname)
+    }
+
+    private fun adjustNickname(nickname: String): String {
+        return if (nickname.length > Nickname.MAX_LENGTH) nickname.substring(
+            0,
+            Nickname.MAX_LENGTH
+        ) else nickname
+    }
```

Comment:
p4) 재활용될 여지가 적고 메서드 내용 자체가 직관적이라 굳이 분리 안해도 괜찮을 것 같아요
```suggestion
        val nickname: String = email.substringBefore(emailDelimiter)

        return if (nickname.length <= Nickname.MAX_LENGTH) nickname
            else nickname.substring(0, Nickname.MAX_LENGTH)
    }
```